src/pyhs3/distributions/composite.py

```diff
@@ -20,47 +21,143 @@ class MixtureDist(Distribution):
+
+        if n_coeffs == n_summands or n_coeffs == n_summands - 1:
+            return coefficients
+
+        msg = (
+            f"Invalid coefficient configuration: {n_coeffs} coefficients "
+            f"for {n_summands} summands. Must have N ({n_summands}) or "
+            f"N-1 ({n_summands - 1}) coefficients."
+        )
+        raise ValueError(msg)
 
     def expression(self, context: Context) -> TensorVar:
         """
         Builds a symbolic expression for the mixture distribution.
 
+        Handles both N and N-1 coefficient cases:
+        - N-1 coefficients: Traditional approach with automatic normalization
+        - N coefficients: Direct summation with optional custom normalization
+
         Args:
             context (dict): Mapping of names to pytensor variables.
 
         Returns:
             pytensor.tensor.variable.TensorVariable: Symbolic representation of the mixture PDF.
         """
+        n_coeffs = len(self.coefficients)
+        n_summands = len(self.summands)
+
+        if n_coeffs == n_summands:
+            # N coefficients case: direct summation with normalization
+            mixturesum = pt.constant(0.0)
+
+            # Calculate the mixture sum
+            for i, coeff in enumerate(self.coefficients):
+                mixturesum += context[coeff] * context[self.summands[i]]
+
+            # Handle normalization
+            if self.ref_coef_norm is not None:
+                # Custom normalization using specified coefficients
+                norm_sum = pt.constant(0.0)
+                for norm_coeff in self.ref_coef_norm:
+                    norm_sum += context[norm_coeff]
+                mixturesum = mixturesum / norm_sum
+            else:
+                # Standard normalization: divide by sum of all coefficients
+                coeffsum = pt.constant(0.0)
+                for coeff in self.coefficients:
+                    coeffsum += context[coeff]
+                mixturesum = mixturesum / coeffsum
+
+        elif n_coeffs == n_summands - 1:
+            # N-1 coefficients case: traditional approach with automatic last term
+            mixturesum = pt.constant(0.0)
+            coeffsum = pt.constant(0.0)
+
+            # Sum the first N-1 terms
+            for i, coeff in enumerate(self.coefficients):
+                coeffsum += context[coeff]
+                mixturesum += context[coeff] * context[self.summands[i]]
+
+            # Add the last term with remaining coefficient
+            last_index = len(self.summands) - 1
+            f_last = context[self.summands[last_index]]
+            mixturesum += (1 - coeffsum) * f_last
+        else:
+            # This should be caught by validation, but included for safety
+            msg = (
+                f"Invalid coefficient configuration: {n_coeffs} coefficients "
+                f"for {n_summands} summands. Must have N or N-1 coefficients."
+            )
+            raise ValueError(msg)
```

Comment:
should remove this, this is basically unreachable code as far as I'm concerned. let's see what codecov says